WordPress/gutenberg
# Quality: Replace wpKebabCase function with kebabCase function from components package

## What?

This PR replaces the `wpKebabCase()` function that was used to generate the slug for a font family with the `kebabCase()` function, which is a private API of the component package.

## Why?

The `kebabCase()` function, which is equivalent to WP Core's _wp_to_kebab_case(), originally existed in the `@wordpress/block-editor` package. In #56758, this function was moved to the `@wordpress/components` package to make it available from various packages.

On the other hand, there is a unique `wpKebabCase()` function that kebabcases the slug of a font when it is uploaded. These two functions have almost the same logic, so we should be able to replace them with the `kebabCase()` function.

## How?

I replaced the functions and removed the unit tests that were no longer needed as a result.

## Testing Instructions

- Open the font library.
- Upload any font that has numbers in the font name.
- Apply the font to the site and save.
- Check that the font is applied on the front end.
- In the generated CSS variables, letters and numbers should be connected with hyphens.
- Confirm that the installed fonts can be deleted.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/19edf54b-2c04-46dd-a895-9014e695c6bf
